pkg/runner/healthcheck_common.go

```diff
@@ -0,0 +1,256 @@
+// container.
+func CustomContainerFixer(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, buildCtx string, containerName string, imageName string, networkID string, portSpecs []string, pull bool, customHostConfig *container.HostConfig, cmds ...string) Fixer {
+	return func() error {
+		_, err := docker.EnsureImage(ctx,
+			log,
+			cli,
+			&docker.BuildImageOpts{
+				Name:     imageName,
+				BuildCtx: buildCtx,
+			})
+		if err != nil {
+			return err
+		}
+		return DefaultContainerFixer(ctx, log, cli, containerName, imageName, networkID, portSpecs, pull, customHostConfig, cmds...)()
+
+	}
+}
+
+// DockerNetworkChecker returns a Checker, a method which when executed will verify a docker network
+// exists with the passed networkID as its name.
+func DockerNetworkChecker(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, networkID string) Checker {
+	return func() bool {
+		networks, err := docker.CheckBridgeNetwork(ctx, log, cli, networkID)
+		if err != nil {
+			log.Error("encountered an error while checking for network %s, %v", networkID, err)
+			return false
+		}
+		return len(networks) > 0
+	}
+}
+
+// DockerNetworkFixer returns a Fixer, a method which when executed will create a docker network
+// with the given name, provided it does not exist already.
+func DockerNetworkFixer(ctx context.Context, log *zap.SugaredLogger, cli *client.Client) Fixer {
+	return func() error {
+		_, err := ensureControlNetwork(ctx, cli, log)
+		return err
+	}
+}
+
+// DialableChecker returns a Checker, a method which when executed will tell us whether a
+// port is dialable. For TCP sockets, a false return could mean the network is unreachable,
+// or that a TCP socket is closed. For UDP sockets, being connectionless, may return a false
+// positive if the network is reachable.
+func DialableChecker(protocol string, address string) Checker {
+	return func() bool {
+		_, err := net.Dial(protocol, address)
+		return err == nil
+	}
+}
+
+// CommandStartFixer returns a Fixer, a method which when executed will start an executable
+// with the given parameters. Uses os/exec to start the command. Cancelling the passed context
+// will stop the executable.
+func CommandStartFixer(ctx context.Context, cmd string, args ...string) Fixer {
+	return func() error {
+		cmd := exec.CommandContext(ctx, cmd, args...)
+		return cmd.Start()
+	}
+}
+
+// DirExistsChecker returns a Checker, a method which when executed will check whether a director
+// exists. A true value means the directory exists. A false value means it does not exist, or
+// that the path does not point to a directory.
+func DirExistsChecker(path string) Checker {
+	return func() bool {
+		fi, err := os.Stat(path)
+		if err != nil {
+			return false
+		}
+		return fi.IsDir()
+	}
+}
+
+// DirExistsFixer returns a Fixer, a method which when executed will create a directory and
+// any parent directories as appropriate.
+func DirExistsFixer(path string) Fixer {
```

Comment:
I really like the util functions. Nice!

Reply:
If only they were all as simple as this one.